packages/ui5-config/src/ui5config.ts

```diff
@@ -334,19 +386,121 @@ export class UI5Config {
+     * @param dataSourcesConfig optional, annotations config that is to be mocked
+     * @param annotationsConfig optional, annotations config that is to be mocked
      * @returns {UI5Config} the UI5Config instance
      * @memberof UI5Config
      */
-    public addMockServerMiddleware(path?: string, annotationsConfig?: MockserverConfig['annotations']): this {
+    public addMockServerMiddleware(
+        appRoot?: string,
+        dataSourcesConfig?: { serviceName: string; servicePath: string }[],
+        annotationsConfig?: MockserverConfig['annotations']
+    ): this {
         this.document.appendTo({
             path: 'server.customMiddleware',
-            value: getMockServerMiddlewareConfig(path, annotationsConfig)
+            value: getMockServerMiddlewareConfig(appRoot, undefined, undefined, dataSourcesConfig, annotationsConfig)
         });
         return this;
     }
 
+    /**
+     * Adds a service configuration to an existing sap-fe-mockserver middleware keeping any existing service configurations. If the config does not contain a sap-fe-mockserver middleware, an error is thrown.
+     *
+     * @param serviceName - name of the service
+     * @param servicePath - path of the service that is to be mocked
+     * @param appRoot - root to the application
+     * @param annotationsConfig optional annotations config that is to be mocked
+     * @returns {UI5Config} the UI5Config instance
+     * @memberof UI5Config
+     */
+    public addServiceToMockserverMiddleware(
+        serviceName: string,
+        servicePath: string,
+        appRoot?: string,
+        annotationsConfig: MockserverConfig['annotations'] = []
+    ): this {
+        const middlewareListYaml = this.document.getSequence({ path: 'server.customMiddleware' });
+        // Throw error if middleware is not found
+        const mockserverMiddlewareYaml = this.document.findItem(
+            middlewareListYaml,
+            (item: any) => item.name === 'sap-fe-mockserver'
+        ) as YAMLMap;
+        if (!mockserverMiddlewareYaml) {
+            throw new Error('Could not find sap-fe-mockserver');
+        } else {
+            // Else append new data to current middleware config and then run middleware update
+            const serviceRoot = `${appRoot ?? './webapp'}/localService/${serviceName}`;
+
+            const mockserverMiddleware = this.findCustomMiddleware('sap-fe-mockserver') as CustomMiddleware;
+            const mockserverMiddlewareConfig = mockserverMiddleware?.configuration as MockserverConfig;
+            if (mockserverMiddlewareConfig.services) {
+                const newServiceData = {
+                    urlPath: servicePath,
+                    metadataPath: `${serviceRoot}/metadata.xml`,
+                    mockdataPath: `${serviceRoot}/data`,
+                    generateMockData: true
+                };
+                const serviceIndex = mockserverMiddlewareConfig.services.findIndex(
```

Comment:
Based on what reasoning are you changing the existing configuration? If I as a developer changed the file by e.g. putting my mockdata somewhere else, how can you decide for me that it must be in a different place?
Just to be clear, adding the new service is fine, manipulating the existing one is what concern me.

Reply:
Initial thoughts were - we should check if the value already exists, otherwise there will be duplicates. But I got the point - we shouldn't update existing one, just do nothing if it already exists.

Done in https://github.com/SAP/open-ux-tools/pull/2499/commits/7958134f87f4d35914573c2bc5e524da9614173b